src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs

```diff
@@ -1,112 +1,159 @@
-            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.", ContainerName, AccountName);
-
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(string.Format("DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}", AccountName, AccountKey));
-            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
-            CloudBlobContainer newContainer = blobClient.GetContainerReference(ContainerName);
+            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.",
+                ContainerName, AccountName);
+            var dt = DateTime.UtcNow;
+            var url = string.Format("https://{0}.blob.core.windows.net/{1}?restype=container", AccountName, ContainerName);
+            StorageUri = String.Format("https://{0}.blob.core.windows.net/{1}/", AccountName, ContainerName);
+            var request = (HttpWebRequest) WebRequest.Create(url);
+            request.Method = "PUT";
+            request.Headers["x-ms-date"] = dt.ToString("R", CultureInfo.InvariantCulture);
+            request.Headers["x-ms-version"] =  AzureHelper.STORAGE_API_VERSION;
+            request.Headers["Authorization"] = AzureHelper.AuthorizationHeader(AccountName, AccountKey, "PUT", dt, request);
 
-            if (FailIfExists && newContainer.Exists())
+            if (ReadOnlyTokenDaysValid > 0)
             {
-                Log.LogError("The container '{0}' already exists.", ContainerName);
-                return false;
+                ReadOnlyToken = AzureHelper.createContainerSASToken(AccountName, ContainerName, AccountKey, "read", ReadOnlyTokenDaysValid);
             }
-            else
+
+            if (WriteOnlyTokenDaysValid > 0)
             {
-                newContainer.CreateIfNotExists();
+                WriteOnlyToken = AzureHelper.createContainerSASToken(AccountName, ContainerName, AccountKey, "write", WriteOnlyTokenDaysValid);
             }
 
-            StorageUri = newContainer.Uri.ToString();
+            Log.LogMessage(MessageImportance.High, "Sending request to create Container");
+            bytestoWrite = new byte[0];
+            bytesToWriteLength = 0;
+            request.BeginGetRequestStream(GetRequestStreamCallback, request);
+            allDone.Reset();
+            allDone.WaitOne();
+            Log.LogMessage(MessageImportance.High, "Received response to create Container");
 
-            // to keep with the msbuild convention of having a path end with
-            // a directory separator char append a '/' to the end of the URI
-            if (!StorageUri.EndsWith("/"))
-                StorageUri = string.Format("{0}/", StorageUri);
+            return true;
+        }
 
-            if (ReadOnlyTokenDaysValid > 0)
+        private static void GetResponseStreamCallback(IAsyncResult callbackResult)
+        {
+            var request = (HttpWebRequest)callbackResult.AsyncState;
```

Comment:
Yeah... don't do this.  It DEFINTELY can't be static (/m will kill your build in most horrible ways). So you need to pass yourself as IAsyncResult.State.

But again, just use the Async stuff instead, it's way cleaner and you don't have to deal with all this messy stuff.